### ISSUE TITLE
[2.3.5] Incorrect html structure after MC-30989

### ISSUE THREAD

**Issue by hostep** [magento/magento2#27920]
### Preconditions (*)
1. Magento 2.3.5 or 2.4-develop (I'm using commmit 1e28f35a6de15f65371c64afda7531611d1be242)

### Steps to reproduce (*)
1. Look at this piece of code: https://github.com/magento/magento2/blob/1e28f35a6de15f65371c64afda7531611d1be242/app/code/Magento/Catalog/view/frontend/templates/product/list/items.phtml#L245-L292
2. Remove a bunch of the inner structure until you have the following remaining:
```php
    <?php if ($showCart):?>
        <?php if ($_item->isSaleable()):?>
            <div class="actions-primary">
        <?php else:?>
        <?php endif; ?>
        </div>
    <?php endif; ?>
```
3. Notice that the opening `<div>` and closing `</div>` aren't in the same block, which can most likely result in incorrect html structure being outputted.

### Expected result (*)
Either:
```php
    <?php if ($showCart):?>
        <?php if ($_item->isSaleable()):?>
            <div class="actions-primary">
            </div>
        <?php else:?>
        <?php endif; ?>
    <?php endif; ?>
```
Or:
```php
    <?php if ($showCart):?>
        <div class="actions-primary">
            <?php if ($_item->isSaleable()):?>
            <?php else:?>
            <?php endif; ?>
        </div>
    <?php endif; ?>
```

If I have to guess, it should probably take the second form, as that's the structure what other templates are following as well:
- https://github.com/magento/magento2/blob/1e28f35a6de15f65371c64afda7531611d1be242/app/code/Magento/Catalog/view/frontend/templates/product/compare/list.phtml#L70-L87
- https://github.com/magento/magento2/blob/1e28f35a6de15f65371c64afda7531611d1be242/app/code/Magento/Catalog/view/frontend/templates/product/list.phtml#L83-L109
- ...

### Actual result (*)
1. See points 2 & 3 of steps to reproduce

### Discussion
This seems to have been introduced in Magento 2.3.5 and 2.4-develop by [MC-30989](https://github.com/magento/magento2/commit/909d4790eea44e57b1c8682149764087d3980653#diff-cbaf433c74e3cb3f5fbdcb8016ccfc1eR246), the `<div class="actions-primary">` was moved inside the `<?php if ($_item->isSaleable()):?>` while it should probably have been left outside of it.

**Comment by sdzhepa** [comment]
PR with fix for this issue has been merged https://github.com/magento/magento2/pull/27926

The Issue can be closed

**Comment by magento-engcom-team** [comment]
Hi @hostep. Thank you for your report. 
The issue has been fixed in magento/magento2#27926 by @hostep in 2.4-develop branch
Related commit(s):
 - [06352384a2e40c246cb7e820b2ef717ca408b935](https://github.com/magento/magento2/commit/06352384a2e40c246cb7e820b2ef717ca408b935)

The fix will be available with the upcoming 2.4.0 release.
